include/seqan3/range/detail/random_access_iterator.hpp

```diff
@@ -270,6 +269,57 @@ class random_access_iterator
     //!\}
 };
 
+/*!\brief A generic random access iterator that delegates most operations to the range.
+ * \tparam container_type The data structure on which the iterator operates, e.g. `std::vector<int>`.
+ * \ingroup range
+ *
+ * The iterator makes certain assumptions on the `container_type`, but does not formally require
+ * it to satisfy the seqan3::random_access_range_concept, because the iterator itself may be
+ * a requirement for this.
+ */
+template <typename container_type>
+class random_access_iterator :
+    public random_access_iterator_base<container_type, random_access_iterator<container_type>>
+{
+private:
+    //!\brief Shortcut for the base class.
+    using base = random_access_iterator_base<container_type, random_access_iterator<container_type>>;
+    //!\brief Import from base class.
+    using typename base::position_type;
+
+public:
+    /*!\name Member types
+     * \brief Make the parent's member types visible.
+     * \{
+     */
+    using typename base::difference_type;
+    using typename base::value_type;
+    using typename base::reference;
+    using typename base::const_reference;
+    using typename base::pointer;
+    using typename base::iterator_category;
+    //!\}
+
+    //!\brief Import the parent's constructors.
+    using base::base;
+    //!\brief Befriend the parent.
+    friend base;
+
+    //!\brief This friend declaration is required to allow non-const to const-construction.
+    template <typename t>
+        requires std::is_const_v<t> && !std::is_const_v<container_type>
+    friend class random_access_iterator;
```

Comment:
I think I should be able to, but I am still convinced that template template is the way to go here. Also because we don't actually want to require the container to be a template!